### PR TITLE
Case 22244: Fix oven crash (RC83)

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/22244/Oven-crashes-and-leaves-some-models-have-an-empty-baked-folder-RC83

Test plan:
- try to bake the models in the original ticket, you won't crash

### CODE REVIEW

File: libraries/render-utils/src/Model.cpp [hunk 1]
```diff
@@ -1572,9 +1572,13 @@ void Model::applyMaterialMapping() {
...
     for (auto& mapping : materialMapping) {
-        std::set<unsigned int> shapeIDs = getMeshIDsFromMaterialID(QString(mapping.first.c_str()));
         auto networkMaterialResource = mapping.second;
-        if (!networkMaterialResource || shapeIDs.size() == 0) {
+        if (!networkMaterialResource) {
+            continue;
+        }
+
+        std::set<unsigned int> shapeIDs = getMeshIDsFromMaterialID(QString(mapping.first.c_str()));
+        if (shapeIDs.size() == 0) {
```
Comment (SamGondelman) [inline libraries/render-utils/src/Model.cpp]: this isn't the fix it's just slightly better

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18747/HighFidelity-Beta-PR15396-2d5cfbf.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18747/HighFidelity-Beta-Interface-PR15396-2d5cfbf.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18747/HighFidelity-Beta-PR15396-2d5cfbf.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18747/HighFidelity-Quest-Beta-PR15396-2d5cfbf.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18747/HighFidelity-Beta-PR15396-2d5cfbf.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18747/HighFidelity-Beta-Interface-PR15396-2d5cfbf.exe

**Pull request merged**